### ISSUE TITLE
When set num_heads = 1, doesnt train

### ISSUE THREAD

**Issue by abeyang00** [PeizeSun/SparseR-CNN#68]
First of all, thank you for the wonderful work. 
I have a quick question though.

I see you have set num_head  = 6, meaning you iteratively update bounding box results 6 times before getting final output.

I wanted to see if i can just use 1 iteration so i set cfg.model.sparsercnn.num_heads = 1

But training does not go well and i get following result

![Screenshot from 2021-02-22 10-57-30](https://user-images.githubusercontent.com/36392866/108647757-ce7fe180-74fc-11eb-811f-57c33e4d76f0.png)

AP only goes upto 3%...

Any suggestion to what i'm doing wrong? should i also set 
cfg.model.sparsercnn.deep_supervision = false? even though this is same as setting num_heads = 1?

Thank you!

**Comment by PeizeSun** [comment]
Hi~
Do you change any other codes?

**Comment by abeyang00** [comment]
@PeizeSun no i didnt change anything else. when i set num_heads=6, it trains well. 

Its only when i set num_heads=1, it doesnt train well

**Comment by PeizeSun** [comment]
when you set num_heads=6, is the final performance the same as our reported performance?

**Comment by abeyang00** [comment]
In the paper AP is 42.3 but i reproduced 38%. Not the same but close enough.

**Comment by PeizeSun** [comment]
Do you train on 8GPUs?

**Comment by abeyang00** [comment]
No, just 1 gpu.

**Comment by PeizeSun** [comment]
I guess only 1 gpu is the reason for 1 head not working.

**Comment by abeyang00** [comment]
so if i only have 1 gpu, i can't train the network if i'm using 1 iteration??

**Comment by PeizeSun** [comment]
Maybe reducing learning rate could help.

**Comment by abeyang00** [comment]
thank you, i'll try it.

**Comment by QinghangHong1** [comment]
> First of all, thank you for the wonderful work.
> I have a quick question though.
> 
> I see you have set num_head = 6, meaning you iteratively update bounding box results 6 times before getting final output.
> 
> I wanted to see if i can just use 1 iteration so i set cfg.model.sparsercnn.num_heads = 1
> 
> But training does not go well and i get following result
> 
> ![Screenshot from 2021-02-22 10-59-26](https://user-images.githubusercontent.com/36392866/108647876-0f77f600-74fd-11eb-84ca-1812f2be862b.png)
> 
> AP only goes upto 3%...
> 
> Any suggestion to what i'm doing wrong? should i also set
> cfg.model.sparsercnn.deep_supervision = false? even though this is same as setting num_heads = 1?
> 
> Thank you!

Hi,

Your plots look really nice and clean. Could you share how you plot the mAP trend from the log, please? Thank you a lot!

**Comment by ZhuYuChenNO1** [comment]
> ts look r

detectron2 will generate tensorboard itself in your output floder, just use it...

**Comment by ZhuYuChenNO1** [comment]
detectron2 will generate tensorboard itself in your output floder, just use it...
